cmd/ingester/confluence/lib/confluence.go

```diff
@@ -46,71 +45,63 @@ type Content struct {
 }
 
 type ConfluenceService struct {
-	authMethod     AuthMethod
-	client         *http.Client
-	hostname       string
-	ingesterClient pb.IngesterClient
+	authMethod AuthMethod
+	client     *http.Client
+	hostname   string
 }
 
-func NewConfluence(hostname string, auth AuthMethod, ingesterClient *pb.IngesterClient) (*ConfluenceService, error) {
+func NewConfluence(hostname string, auth AuthMethod) (*ConfluenceService, error) {
 	return &ConfluenceService{
-		authMethod:     auth,
-		client:         &http.Client{},
-		hostname:       hostname,
-		ingesterClient: *ingesterClient,
+		authMethod: auth,
+		client:     &http.Client{},
+		hostname:   hostname,
 	}, nil
 }
 
-func (c *ConfluenceService) Fetch(modifiedSince time.Time) error {
-	stream, err := c.ingesterClient.Ingest(context.Background())
-	if err != nil {
-		return err
-	}
-
+func (c *ConfluenceService) Fetch(modifiedSince time.Time) ([]*pb.Document, error) {
+	var allDocuments []*pb.Document
 	startIdx := 0
 	for {
 		endpoint, err := c.constructEndpoint(startIdx)
 		if err != nil {
-			return err
+			return allDocuments, err
 		}
 		req, err := http.NewRequest("GET", endpoint, nil)
 		if err != nil {
-			return err
+			return allDocuments, err
```

Comment:
nit: probably just return `nil` instead of `allDocuments` here and on lines 76 and 81